lwt/unix_fcntl_lwt_stubs.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2016 Jeremy Yallop <yallop@docker.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *
+ */
+
+#include <errno.h>
+#include <string.h>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include "caml/memory.h"
+#include "caml/alloc.h"
+
+#include "lwt_unix.h"
+
+struct job_open {
+  struct lwt_unix_job job;
+  int flags;
+  int has_perms;
+  int perms;
+  int fd;
+  int error_code;
+  char *name;
+  char data[];
+};
+
+
+static void worker_open(struct job_open *job)
+{
+  job->fd = job->has_perms
+    ? open(job->name, job->flags, job->perms)
+    : open(job->name, job->flags);
+  job->error_code = errno;
+}
+
+static value result_open(struct job_open *job)
+{
+  LWT_UNIX_CHECK_JOB_ARG(job, job->fd < 0, "open", job->name);
+  lwt_unix_free_job(&job->job);
```

Comment:
I think this frees the job twice? See <https://github.com/ocsigen/lwt/blob/2084cff47c1bea5efb6a67b5f960bf1982859f43/src/unix/lwt_unix.h#L290>.